SimplenoteTests/NoticeControllerTests.swift

```diff
@@ -0,0 +1,238 @@
+
+class NoticeControllerTests: XCTestCase {
+    private lazy var presenter = MockNoticePresenter()
+    private lazy var timerFactory = MockTimerFactory()
+    private lazy var controller = NoticeController(presenter: presenter, timerFactory: timerFactory)
+
+    func testSetupNoticeContoller() {
+        controller.setupNoticeController()
+
+        XCTAssertTrue(presenter.listeningToKeyboard == true)
+    }
+
+    func testPresentNotice() {
+        let notice = Notice(message: "Message", action: nil)
+        let expectedActions: [MockNoticePresenter.Action] = [
+            .present(notice.message)
+        ]
+        controller.present(notice)
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertEqual(presenter.lastNoticeView?.message, notice.message)
+    }
+
+    func testControlDoesNotPresentNewNoticeIfPresenting() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+        let expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        controller.present(noticeA)
+        controller.present(noticeB)
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertEqual(presenter.lastNoticeView?.message, noticeA.message)
+    }
+
+    func testAppendToQueueIfNew() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
+
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
+
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
+    }
+
+    func testMakeNoticeView() throws {
+        let notice = Notice(message: "Message", action: nil)
+
+        controller.present(notice)
+
+        XCTAssertEqual(notice.message, presenter.lastNoticeView?.message)
+        XCTAssertNil(presenter.lastNoticeView?.handler)
+    }
+
+    func testMakeNoticeViewWithAction() {
+        let action = NoticeAction(title: "Title") {
+            print("Action")
+        }
+        let notice = Notice(message: "Message ", action: action)
+
+        controller.present(notice)
+
+        XCTAssertEqual(notice.message, presenter.lastNoticeView?.message)
+        XCTAssertEqual(notice.action?.title, presenter.lastNoticeView?.actionTitle)
+        XCTAssertNotNil(presenter.lastNoticeView?.handler)
+    }
+
+    func testMakeNoticeViewWithEmptyNotice() {
+        let action = NoticeAction(title: "") { }
+        let notice = Notice(message: "", action: action)
+
+        controller.present(notice)
+
+        XCTAssertEqual(notice.message, presenter.lastNoticeView?.message)
+        XCTAssertNotEqual(notice.action?.title, presenter.lastNoticeView?.actionTitle)
```

Comment:
To continue the previous conversation, we can update actionTitle getter in NoticeView to `noticeButton.title(for: .normal)`. That would make it behave the same as `testMakeNoticeViewWithAction`